wd_helper.au3

```diff
@@ -449,6 +449,78 @@ Func _WD_WaitElement($sSession, $sStrategy, $sSelector, $iDelay = Default, $iTim
 	Return SetError(__WD_Error($sFuncName, $iErr, $sParameters), 0, $sElement)
 EndFunc   ;==>_WD_WaitElement
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_WaitScript
+; Description ...: Wait for a JavaScript snippet to return true.
+; Syntax ........: _WD_WaitScript($sSession, $sJavaScript[, $iDelay = Default[, $iTimeout = Default[, $iOptions = Default]]])
+; Parameters ....: $sSession  - Session ID from _WD_CreateSession
+;                  $sJavaScript - JavaScript to run
+;                  $iDelay    - [optional] Milliseconds to wait before initially checking status
+;                  $iTimeout  - [optional] Period of time (in milliseconds) to wait before exiting function
+; Return values .: Success - True
+;                  Failure - False and sets @error to one of the following values:
+;                  - $_WD_ERROR_InvalidArgue
+;                  - $_WD_ERROR_RetValue
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_Timeout
+;                  - $_WD_ERROR_UserAbort
+; Author ........: yehiaserag
+; Modified ......:
+; Remarks .......: The Javascript needs to return either True or False.
+; Related .......: _WD_ExecuteScript
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_WaitScript($sSession, $sJavaScript, $iDelay = Default, $iTimeout = Default)
+	Local Const $sFuncName = "_WD_WaitScript"
+	Local Const $sParameters = 'Parameters:   JavaScript=' & $sJavaScript & '   Delay=' & $iDelay & '   Timeout=' & $iTimeout
+	Local $iErr
+	Local $bValue = False
+
+	If $iDelay = Default Then $iDelay = 0
+	If $iTimeout = Default Then $iTimeout = $_WD_DefaultTimeout
+
+	If StringLeft($sJavaScript, 6) <> "return" Then
+		$iErr = $_WD_ERROR_InvalidArgue
+	Else
+		__WD_Sleep($iDelay)
+		$iErr = @error
+
+		; prevent multiple errors https://github.com/Danp2/au3WebDriver/pull/290#issuecomment-1100707095
+		Local $_WD_DEBUG_Saved = $_WD_DEBUG ; save current DEBUG level
+
+		; Prevent logging from _WD_ExecuteScript if not in Full debug mode
+		If $_WD_DEBUG <> $_WD_DEBUG_Full Then $_WD_DEBUG = $_WD_DEBUG_None
+
+		Local $hWaitTimer = TimerInit()
+		While 1
+			If $iErr Then ExitLoop
```

Comment:
The check is at the beginning so that it will also work for the __WD_Sleep that occurs prior to the start of the loop. 

Seems like we've discussed this previously for one of the other similar functions. 🤔

Reply:
Yes I know. I said already that I see that.
I'm not refering to `__WD_Sleep` here.
I'm wondering about refactoring this:
```autoit
While 1
	If $iErr Then ExitLoop
	.....
WEnd
```
to this form
```autoit
Do
	.....
Until $iErr
```

it looks more obvius for me to check `$iErr` on the end not at the begining.

Reply:
about the fact that `I'm nitpicky sometimes.`
?
Yes I know.
If so then you can consider this part as resolved.

Reply:
> If the loop is restructured to a Do...While, then you would need to also add code so that the loop is never entered if a user abort occurred.

I plan to propose some tweaks (PR) for this.
ASAP

Reply:
Why? ☹️

Reply:
After rethinking we should leave it here as it is because this following error checking
https://github.com/ye7iaserag/au3WebDriver/blob/master/wd_helper.au3#L496-L497

should be at start to catchup this error:
https://github.com/ye7iaserag/au3WebDriver/blob/master/wd_helper.au3#L486-L487

Reply:
Glad we are finally in agreement on this. 😄

Reply:
this part should be considered as resolved.